aca_hi_bgd/update_bgd_events.py

```diff
@@ -1198,11 +1229,32 @@ def main(args=None):  # noqa: PLR0912, PLR0915 too many branches, too many state
             ]
         start = CxoTime(opt.start)
     if start is None:
-        start = CxoTime(-7)
-
-    new_events, last_proc_time = get_events(
-        start, stop=opt.stop, outdir=opt.web_out, data_root=opt.data_root
-    )
+        start = CxoTime.now() - 7 * u.day
+
+    # Figure out what range of data we can actually process
+    if opt.maude is True:
+        # get the end of backorbit data from maude.  Put a retry around this
+        # in case of intermittent failures.
+        with maude_conf.set_temp("timeout", 5):
+            last_telem_date = retry_call(
+                get_last_backorbit_date, [["AACCCDPT", "AOIMAGE0"]], tries=5, delay=5
+            )
+        last_telem_date = CxoTime(last_telem_date)
```

Comment:
This is another example of nice code that should be upstreamed, in this case to `fetch.get_time_range`. That function would use the current `data_source` to decide how to get the time range. The calling code here would look like:
```
msids_time_range = ["AACCCDPT"]
if opt.maude:
    msids_time_range, append("AOIMAGE0")
_, data_stop = fetch.get_time_range(msids_time_range)
```
The asymmetry does make we wonder what happens for the `cxc` data source if the L0 images do not extend through the end of available `AACCCDPT` telemetry.

Reply:
I haven't made the upstream changes in cheta for get_time_range, but I think with the updates to maude on get_last_backorbit_date that this is done enough for now.